### PR TITLE
[STYLE] Drop python 3.7, linting applied and improvements of it

### PR DESCRIPTION
**Pull request by Mildophin**
This MR is mostly about linting following my previous MR as it wasn't applied correctly. This MR also includes the changes of https://github.com/pasqal-io/pasqal-cloud/pull/88, with some updates of the linting dependencies, the files of the CI and pre commit hooks. There are few things I forgot to change in my last MR, so this version patch them.

The MR is a fork of the following one: https://github.com/pasqal-io/pasqal-cloud/pull/95 (you find some comments from aleksander on it)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -24,12 +22,13 @@ repos:
...
+    rev: v2.2.0
     hooks:
       - id: autoflake
   - repo: https://github.com/PyCQA/flake8
     rev: 6.0.0
     hooks:
       - id: flake8
-        exclude: setup.py
-        language_version: python3.11
+        exclude: ^sdk/
```
Comment (Mildophin) [inline .pre-commit-config.yaml:32]: to prevent non used imports errors being raised

File: pasqal_cloud/__init__.py [hunk 2]
```diff
@@ -20,7 +20,12 @@
 from pasqal_cloud.batch import Batch, RESULT_POLLING_INTERVAL
 from pasqal_cloud.client import Client
 from pasqal_cloud.device import BaseConfig, EmulatorType
-from pasqal_cloud.endpoints import AUTH0_CONFIG, Auth0Conf, Endpoints, PASQAL_ENDPOINTS
+from pasqal_cloud.endpoints import (  # noqa: F401
```
Comment (Mildophin) [inline pasqal_cloud/__init__.py:23]: as the endpoints and auth0 config is not used directly here but imported by the users, using the noqa flag prevent flake8 errors

File: pasqal_cloud/_version.py [hunk 3]
```diff
@@ -12,4 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "0.3.3"
+__version__ = "0.3.4"
```
Comment (MatthieuMoreau0) [inline pasqal_cloud/_version.py:15]: I don't think this warrants a version bump tbh

**Review by MatthieuMoreau0 — COMMENTED** [review]
lgtm, two small comments

File: pasqal_cloud/_version.py (hunk 3 above)
Comment (Mildophin) [reply pasqal_cloud/_version.py:15]: Alright, I had a doubt regarding the python version dependency

**Review by MatthieuMoreau0 — APPROVED** [review]

**Pull request merged**